src/class-metadata.php

```diff
@@ -102,12 +102,14 @@ public function construct_metadata( array $parsely_options, WP_Post $post ): arr
 			$authors  = $this->get_author_names( $post );
 			$category = $this->get_category_name( $post, $parsely_options );
 
-			if ( has_post_thumbnail( $post ) ) {
-				$image_id  = get_post_thumbnail_id( $post );
-				$image_url = wp_get_attachment_image_src( $image_id );
-				$image_url = $image_url[0];
-			} else {
-				$image_url = $this->get_first_image( $post );
+			// Get featured image and thumbnail.
+			$image_url = get_the_post_thumbnail_url( $post, 'full' );
+			if ( false === $image_url ) {
```

Comment:
`get_the_post_thumbnail_url()` has a late filter, so it could return any data type (like `null`) if someone applies a mis-configured filter.

So that we're definitely dealing with a string, then consider change this check (and the instance below) to:

```suggestion
			if ( ! is_string( $image_url ) ) {
```

That way, we know that `null`, `true`, `5` and other non-strings can't get through, while still guarding against the `false` which is the most likely possibility.

It also makes the code a little more agnostic. My suggestion basically says "If it's not a string, then make it an empty string", where as the code you've got says "If it is a literal false (because `get_the_post_thumbnail_url()` couldn't find a truthy ID from the `$post`), then make it an empty string". i.e. we want to test for the actual thing we're looking for (a string), and not just a specific other data type that we presume the function might return.

Reply:
Exceptional catch. Implemented.